### ISSUE TITLE
Realm file size limitations

### ISSUE THREAD

**Issue by ghost** [realm/realm-dotnet#536]
Hi,

in the documentation it says for iOS

> Any single Realm file cannot be larger than the amount of memory your application would be allowed to map in iOS — this changes per device, and depends on how fragmented the memory space is at that point in time (there is a radar open about this issue: rdar://17119975). If you need to store more data, you can map it over multiple Realm files.

In layman terms I am assuming the realm file is the equivalent of the database

So based on the example the real realm file/database can only be about 2GB in size, even then it dependant on the state of the device.

Is my assumption correct? If so how does realm warn the developer of this, so that they can create a new realm file. Is it also true to say if the data had to be split between 2 realm files/databases, of 2GB each only one could be open at a time?

Would it be possible to update the documentation to clarify this limitation. I feel this is something developers would need to be sure about and understand before adopting realm, especially if they want to store binary data like images and video in a realm file/database

**Comment by bmunkholm** [comment]
Yeah - well "only" 2 GB is usually a lot even on todays mobile devices. I doubt any user would be happy if your app uses that much space anyway. No database is really a good fit for big blobs anyway. You should always store those as separate files and just store the filepath in the database. What's good to put in the database is data you want to query. Hopefully that should remove your needs to really have that big databases?
Today there is no real warning if you use up all the available space. The best recommendation is to monitor free disk space yourself.
We will update the docs to make it more clear - thanks for the feedback!

**Comment by cbush** [comment]
@bmunkholm is this still relevant?

**Comment by bmunkholm** [comment]
I can't if the docs got updated on this. But it's still relevant to warn users against storing big blobs (e.g. images and video), and only use the DB for meta data.
